src/shared/components/NotFound.tsx

```diff
@@ -136,10 +140,18 @@ const NotFound: FC = () => {
 
   const handleDeepLink = useCallback(async () => {
     if (!org.current) {
-      setIsFetchingOrg(true)
-      org.current = await fetchOrg()
-    }
+      if (shouldUseQuartzIdentity()) {
+        setIsFetchingOrg(true)
+        const defaultAccount = await getDefaultAccount()
```

Comment:
Once you implement the change Bucky's suggesting to the function in the API layer, this will have the potential to throw an error if there's a problem. At that point, I think you'll need to add a `try/catch` block here, and then any eventing or other error handling can happen in the `catch`.

Reply:
okay that a really good idea. Is there some other way we can handle error other than `console.error()` here? 🤔